test/src/tests/acceptancetests/lrogrouptest/lro_test.go

```diff
@@ -23,72 +24,638 @@ var lrosClient = getLROsClient()
+	c.Assert(err, chk.NotNil)
+}
+
+func (s *LROSuite) TestSADsPutAsyncRelativeRetry400(c *chk.C) {
+	future, err := lroSADSClient.PutAsyncRelativeRetry400(context.Background(), &lrogroup.Product{})
+	c.Assert(err, chk.IsNil)
+	err = future.WaitForCompletionRef(context.Background(), lroSADSClient.Client)
+	c.Assert(err, chk.NotNil)
+}
+
+func (s *LROSuite) TestSADsPutAsyncRelativeRetryInvalidHeader(c *chk.C) {
+	future, err := lroSADSClient.PutAsyncRelativeRetryInvalidHeader(context.Background(), &lrogroup.Product{})
+	c.Assert(err, chk.IsNil)
+	err = future.WaitForCompletionRef(context.Background(), lroSADSClient.Client)
+	c.Assert(err, chk.NotNil)
+}
+
+func (s *LROSuite) TestSADsPutAsyncRelativeRetryInvalidJSONPolling(c *chk.C) {
+	future, err := lroSADSClient.PutAsyncRelativeRetryInvalidJSONPolling(context.Background(), &lrogroup.Product{})
+	c.Assert(err, chk.IsNil)
+	err = future.WaitForCompletionRef(context.Background(), lroSADSClient.Client)
+	c.Assert(err, chk.NotNil)
+}
+
+func (s *LROSuite) TestSADsPutAsyncRelativeRetryNoStatus(c *chk.C) {
+	future, err := lroSADSClient.PutAsyncRelativeRetryNoStatus(context.Background(), &lrogroup.Product{})
+	c.Assert(err, chk.IsNil)
+	err = future.WaitForCompletionRef(context.Background(), lroSADSClient.Client)
+	c.Assert(err, chk.NotNil)
+}
+
+func (s *LROSuite) TestSADsPutAsyncRelativeRetryNoStatusPayload(c *chk.C) {
+	future, err := lroSADSClient.PutAsyncRelativeRetryNoStatusPayload(context.Background(), &lrogroup.Product{})
+	c.Assert(err, chk.IsNil)
+	err = future.WaitForCompletionRef(context.Background(), lroSADSClient.Client)
+	c.Assert(err, chk.NotNil)
+}
+
+func (s *LROSuite) TestSADsPutError201NoProvisioningStatePayload(c *chk.C) {
+	future, err := lroSADSClient.PutError201NoProvisioningStatePayload(context.Background(), &lrogroup.Product{})
+	c.Assert(err, chk.IsNil)
+	_, err = future.Result(lroSADSClient)
+	c.Assert(err, chk.NotNil)
+}
+
+func (s *LROSuite) TestSADsPutNonRetry201Creating400(c *chk.C) {
+	future, err := lroSADSClient.PutNonRetry201Creating400(context.Background(), &lrogroup.Product{})
+	c.Assert(err, chk.IsNil)
+	err = future.WaitForCompletionRef(context.Background(), lroSADSClient.Client)
+	c.Assert(err, chk.NotNil)
+}
+
+func (s *LROSuite) TestSADsPutNonRetry201Creating400InvalidJSON(c *chk.C) {
+	future, err := lroSADSClient.PutNonRetry201Creating400InvalidJSON(context.Background(), &lrogroup.Product{})
+	c.Assert(err, chk.IsNil)
+	err = future.WaitForCompletionRef(context.Background(), lroSADSClient.Client)
+	c.Assert(err, chk.NotNil)
+}
+
+func (s *LROSuite) TestSADsPutNonRetry400(c *chk.C) {
+	_, err := lroSADSClient.PutNonRetry400(context.Background(), &lrogroup.Product{})
+	c.Assert(err, chk.NotNil)
+}
+
+// custom header client
+
+func (s *LROSuite) TestCustomHeaderPost202Retry200(c *chk.C) {
+	/*req, err := lroCustomHeaderClient.Post202Retry200Preparer(context.Background(), &lrogroup.Product{})
```

Comment:
is there a problem with this one ? ( and the next ones) if so maybe create an issue for them?

Reply:
Thanks I forgot about this, will update the comment and open an issue.

Reply:
Opened issue https://github.com/Azure/azure-sdk-for-go/issues/1922 to track adding custom header data.